graph/step.go

```diff
@@ -196,3 +202,42 @@ func (s *Step) UpdateBuildStepWithDefaults() {
 		s.Build = fmt.Sprintf("--isolation hyperv %s", s.Build)
 	}
 }
+
+// UseBuildCacheForBuildStep indicates if buildx needs to be used
+func (s *Step) UseBuildCacheForBuildStep() bool {
+	if s.EnableCache && s.CacheID == "" {
+		s.CacheID = "cache"
+	}
+
+	return s.IsBuildStep() && s.CacheID != ""
+}
+
+// FetchBuildWithBuildCache returns an updated build step by adding buildx cache parameters
+func (s *Step) FetchBuildWithBuildCache() (string, error) {
```

Comment:
Method name could be improved since it doesn't do anything related to its description

Reply:
`GetCmdWithCacheFlags()` ?